test/gtest/aca_test_dhcp.cpp

```diff
@@ -180,3 +231,369 @@ TEST(dhcp_message_test_cases, get_options_valid)
   retcode = ACA_Dhcp_Server::get_instance()._get_requested_ip(&stDhcpMsg);
   EXPECT_EQ(retcode, 0x0a000001);
 }
+
+TEST(dhcp_request_test_case, DISABLED_l2_dhcp_test)
+{
+  ulong not_care_culminative_time = 0;
+  string cmd_string;
+  int overall_rc;
+
+  // create and setup br-int and br-tun bridges, and their patch ports
+  overall_rc = ACA_OVS_L2_Programmer::get_instance().setup_ovs_bridges_if_need();
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // re adding dhcp default flows
+  ACA_OVS_L2_Programmer::get_instance().execute_openflow_command(
+          "add-flow br-int \"table=0,priority=25,udp,udp_src=68,udp_dst=67,actions=CONTROLLER\"",
+          not_care_culminative_time, overall_rc);
+  ASSERT_EQ(overall_rc, EXIT_SUCCESS);
+
+  // monitor br-int for dhcp request message
+  ovs_monitor_thread = 
+    new thread(bind(&ACA_OVS_Control::monitor, &ACA_OVS_Control::get_instance(), "br-int", "resume"));
+  ovs_monitor_thread->detach();
```

Comment:
see if you can try not to detach so that you can join the thread at the end of the test case, and then kill that thread. Let me know if that works.

Reply:
The thread will block the test case because of the monitor thread does not exit itself.